package guild;
option py_generic_services=true;
import "universal/public.proto";
import "common.proto";

//职位
enum position{
	TYPE_CHAIRMAN=1;//帮主
	TYPE_CHAIRMAN_VICE=2;//副帮主
	TYPE_ELDER=3;//长老
	TYPE_ELITE=4;//精英
	TYPE_COMMON=5;//帮众
	TYPE_APPRENTICE=6;//学徒
}

//帮派建筑
enum build{
	BUILD_MAIN=1;//聚义厅
	BUILD_WAREHOUSE=2;//库房
	BUILD_WING=3;//厢房
	BUILD_TRAIN=4;//训练场
	BUILD_COIN=5;//铸币场
	BUILD_PET=6;//异兽房
}

//帮派操作
enum operate{
	OPERATE_CREATE=1;//创建帮派
	OPERATE_JOIN=2;//加入帮派
	OPERATE_QUIT=3;//退出帮派
	OPERATE_TENET=4;//修改宗旨
	OPERATE_NOTICE=5;//修改公告
	OPERATE_MAIL=6;//帮派邮件
	OPERATE_JOIN_CANCEL=7;//取消加入帮派
	OPERATE_UPGRADE=8;//升级帮派建筑
}

//成员信息
message member {
	required int64 roleId=1;	//角色ID
	optional int32 shape=2; //造型
	optional int32 level=3;	//等级
	optional int32 school=4;	//职业
	optional bytes name=5;	//名字
	optional position position=6;//职位
	optional int32 curContri=7;	//当前贡献
	optional int32 totalContri=8;	//历史贡献
	optional int32 fightCnt=9;	//帮战次数
	optional int32 activeDays=10;	//活跃天数
	optional int32 joinTime=11;	//入帮时间
	optional int32 offlineTime=12;	//离线时间,0表示在线
	repeated int32 shapeParts=13; //造型部位，参照role.roleAttr
	repeated int32 colors=14; // 染色, 参照role.roleAttr
	repeated int32 guildTask=15; // 本周任务[完成次数,总次数]
	repeated int32 guildFight=16; // 本周帮战[参加次数,总次数]
	optional bool banTalk=17; //是否禁言中
}

//创建信息
message createGuild{
	optional bytes name=1;//名字
	optional bytes tenet=2;//宗旨
}

//帮派成员分页信息
message guildMember{
	repeated member members=1;//工会成员
	optional int32 page=2;//当前页
	optional int32 totalPage=3;//总页数
}

//建筑信息
message buildMsg{
	required build idx=1;	//建筑序号
	optional int32 level=2;	//建筑等级
	optional int32 cost=3;	//升级消耗资金
	optional int32 need=4;	//升级需求资金
	optional int32 time=5;	//升级剩余时间
	optional bool top=6;	//是否最高级
}

//建筑列表
message buildList{
	repeated buildMsg build=1;
}


//申请信息
message applyInfo{
	required int64 roleId=1;//角色ID
	optional int32 level=2;//等级
	optional int32 school=3;//门派
	optional bytes name=4;//名字
}

//申请列表
message applyList{
	repeated applyInfo applys=1;
}

//事件列表
message logList{
	repeated bytes log=18;//日志
}

//帮派详细信息
message guildInfo{
	required int32 guildId=1; //帮派ID
	optional int32 level=2; //帮派等级
	optional int32 vitality=3;	//活跃度
	optional int32 fund=4; //资金
	optional int32 maintain=5; //维护资金
	optional int32 bonus=6;//分红
	optional int32 online=7;//在线人数
	optional int32 size=8;//总人数
	optional int32 maxSize=9;//人数上限
	optional int32 birth=10;//建立时间:UTC秒
	optional bytes name=11;//帮派名字
	optional bytes chairman=12;//现任帮主
	optional bytes notice=13;//公告
	optional bytes tenet=14;//宗旨
	optional guildMember member=15;//成员列表
	optional applyList apply=16;//申请列表
	optional buildList build=17;//建筑列表
	optional logList log=18;//事件列表
	optional int32 apprenticeOnline=19;//学徒在线人数
	optional int32 apprenticeSize=20;//总学徒数
	optional int32 apprenticeMax=21;//学徒上限
	optional int64 chairmanId=22;//盟主ID
}

//帮派列表信息
message guildOverview{
	required int32 guildId=1;	//帮派ID
	optional int32 level=2;	//帮派等级
	optional int32 vitality=3;	//活跃度
	optional int32 size=4;	//人数
	optional int32 maxSize=5;	//人数上限
	optional bytes name=6;	//帮派名字
	optional bytes tenet=7;	//帮派宗旨
	optional member chairman=8;	//帮主信息
	optional bool isApply=9;	//是否申请过:true申请了,false未申请
}

//帮派列表
message guildList{
	repeated guildOverview guilds=1;
}

//操作信息
message operateInfo{
	required operate idx=1;	//操作序号
	optional bool result=2;	//操作结果true成功;false失败
	optional int32 args=3;	//操作参数
}

//任命
message appointInfo{
	required int64 roleId=1;//角色ID
	optional position position=2;//职位
}

//福利
message welfareInfo{
	optional int32 bonus=1;//分红
	optional bool isBonusGet=2;//分红是否已领取:true已领;false未领
}

//成员通知
message memberNotify{
	required int32 idx=1;	//通知类型1增加2删除3修改
	optional member member=2;	//成员信息
}

//申请通知
message applyNotify{
	required int32 idx=1;	//通知类型1清空2删除
	optional int64 roleId=2;//申请人ID
}


//对手匹配
message pkInfo{
	repeated bytes guildNameList = 1; //帮派名
}

//帮战信息
message fightInfo{
	optional int32 state = 1; //帮战阶段状态, 1.报名期 2.公示期 3.入场期 4.开战期
	optional bool isSignUp = 2; //是否已报名
	optional bool isAutoSignUp = 3; //是否自动报名
	optional pkInfo pkList = 4; //对手匹配
}

//帮战精英队队员
message fightTeamMember{
	required int64 roleId = 1; //角色id
	optional bytes name = 2; //角色名
	optional int32 level = 3; //等级
	optional int32 school = 4; //门派
	optional int32 fightPower = 5; //战力
	optional int32 state = 6; //状态, 0.离线 1.在线
	optional int64 offlineTime = 7; //上次离线时间
	optional int32 teamNo = 8; //队号, 0.非精英 1.一队 2.二队 3.三队
	optional bool isAbsent = 9; //是否缺席
	optional int32 shape = 10; //造型
}

//帮战精英队
message fightTeam{
	repeated fightTeamMember memberList = 1; //精英队队员列表
}

//搜索
message searchMsg{
	required bytes keyword = 1; //搜索关键词 
}

//搜索结果
message searchResult{
	repeated fightTeamMember memberList = 1; //搜索到的成员列表
}

//帮战战况
message pkResult{
	optional int64 time = 1; //帮战日期
	optional bytes result = 2; //战果
	optional bytes targetGuildName = 3; //对手帮派名称
	optional bytes score = 4; //比分
	repeated int32 teamResultList = 5; //精英组结果, 0.激战中 1.胜利 2.失败 3.平局
}

//帮战战报
message fightResult{
	repeated pkResult pkResultList = 1; //战况列表
	optional int32 fightCount = 2; //帮战次数
	optional int32 winCount = 3; //胜利场数
}


//上行
service terminal2main{
	rpc rpcGuildRequest(public.fake)returns(public.fake);//请求帮派信息，没帮的人员会收到帮派列表信息，在帮人员会收到帮派详细信息
	rpc rpcGuildCreate(guild.createGuild)returns(public.fake);//创建帮派
	rpc rpcGuildJoin(common.int32_)returns(public.fake);//加入帮派
	rpc rpcGuildJoinCancel(common.int32_)returns(public.fake);//取消加入帮派
	rpc rpcGuildQuit(public.fake)returns(public.fake);//退出帮派
	rpc rpcGuildListReq(public.fake)returns(public.fake);//请求帮派列表
	rpc rpcGuildApplyAll(public.fake)returns(public.fake);//一键申请
	rpc rpcGuildSearch(public.fake)returns(public.fake);//搜索帮派
	rpc rpcGuildBack(common.int32_)returns(public.fake);//回到帮派:0为回自己的帮派，非0为去指定ID的帮派
	rpc rpcGuildDismiss(public.fake)returns(public.fake);//解散帮派
	rpc rpcGuildAppoint(guild.appointInfo)returns(public.fake);//任命
	rpc rpcGuildKick(common.int64_)returns(public.fake);//踢出
	rpc rpcGuildBan(common.int64_)returns(public.fake);//禁言
	rpc rpcGuildUnBan(common.int64_)returns(public.fake);//解禁
	rpc rpcGuildJoinAgree(common.int64_)returns(public.fake);//同意申请
	rpc rpcGuildJoinReject(common.int64_)returns(public.fake);//拒绝申请
	rpc rpcGuildJoinInvite(common.int64_)returns(public.fake);//邀请入帮
	rpc rpcGuildBuildList(public.fake)returns(public.fake);//请求帮派建筑列表
	rpc rpcGuildApplyList(public.fake)returns(public.fake);//请求申请入帮列表
	rpc rpcGuildClearApply(public.fake)returns(public.fake);//清空申请列表
	rpc rpcGuildTenet(common.bytes_)returns(public.fake);//修改帮派宗旨
	rpc rpcGuildNotice(common.bytes_)returns(public.fake);//修改帮派公告
	rpc rpcGuildMail(common.bytes_)returns(public.fake);//群体邮件
	rpc rpcGuildAdvertise(public.fake)returns(public.fake);//帮派宣传
	rpc rpcGuildBuildUp(guild.buildMsg)returns(public.fake);//升级建筑
	rpc rpcGuildBonusGet(public.fake)returns(public.fake);//领取分红
	
	//帮战相关
	rpc rpcGuildFightInfoRequest(public.fake)returns(public.fake);//请求帮战信息
	rpc rpcGuildFightAutoSignUpSet(common.bool_)returns(public.fake);//设置自动报名, false.取消自动报名  true.确定自动报名
	rpc rpcGuildFightSignUp(public.fake)returns(public.fake);//帮战报名
	rpc rpcGuildFightTeamRequest(public.fake)returns(public.fake);//请求帮战精英队
	rpc rpcGuildFightTeamMemberSelect(guild.fightTeamMember)returns(public.fake);//选择精英
	rpc rpcGuildFightTeamMemberCancel(guild.fightTeamMember)returns(public.fake);//取消精英
	rpc rpcGuildFightSearch(guild.searchMsg)returns(public.fake);//精英搜索
	rpc rpcGuildFightResultRequest(public.fake)returns(public.fake);//请求帮战战报
}

//下行
service main2terminal{
	rpc rpcGuildList(guild.guildList)returns(public.fake);//帮派列表
	rpc rpcGuildInfo(guild.guildInfo)returns(public.fake);//帮派详细信息
	rpc rpcGuildInfoUpdate(guild.guildInfo)returns(public.fake);//更新帮派信息(只更新有的字段)
	rpc rpcGuildMemberNotify(guild.memberNotify)returns(public.fake);//帮派成员变动通知
	rpc rpcGuildApplyNotify(guild.applyNotify)returns(public.fake);//帮派申请变动通知
	rpc rpcGuildApplyInfo(guild.applyList)returns(public.fake);//申请入帮信息
	rpc rpcGuildBuildInfo(guild.buildList)returns(public.fake);//帮派建筑信息
	rpc rpcGuildOperateRes(guild.operateInfo)returns(public.fake);//帮派相关操作结果
	rpc rpcGuildWelfareInfo(guild.welfareInfo)returns(public.fake);//帮派福利信息

	//帮战相关
	rpc rpcGuildFightInfo(guild.fightInfo)returns(public.fake);//帮战信息
	rpc rpcGuildFightInfoChange(guild.fightInfo)returns(public.fake);//修改帮战信息
	rpc rpcGuildFightTeam(guild.fightTeam)returns(public.fake);//帮战精英队
	rpc rpcGuildFightTeamMemberAdd(guild.fightTeamMember)returns(public.fake);//增加精英
	rpc rpcGuildFightTeamMemberDelete(guild.fightTeamMember)returns(public.fake);//删除精英
	rpc rpcGuildFightSearchResult(guild.searchResult)returns(public.fake);//精英搜索结果
	rpc rpcGuildFightResult(guild.fightResult)returns(public.fake);//帮战战报
}
